db/migration/migration.go

```diff
@@ -105,21 +105,51 @@ func Run(subCmd string, args []string) {
 	}
 
 	if err = goose.LoadMigrationPlugins(migrationsPath); err != nil {
-		logger.Error("migration: failed to load migration plugins: %s", err)
-		return
+		fmt.Printf("error: failed to load migration plugins: %s\n", err)
+		os.Exit(1)
 	}
 
-	if err != nil {
-		logger.Error("migration: failed to load runtime migrations: %s", err.Error())
-		return
+	if err := lock(dbType, db); err != nil {
```

Comment:
defer unlock. and, by the way: https://github.com/cloudwan/goose/pull/4#discussion_r109954692